language-extensions/dotnet-core-CSharp/src/managed/utils/Logging.cs

```diff
@@ -41,7 +54,111 @@ public static void Trace(string message)
         /// </param>
         public static void Error(string message)
         {
-            Console.Error.WriteLine(message);
+            try
+            {
+                Console.Error.WriteLine(message);
+            }
+            catch (Exception)
+            {
+                // Ignore exceptions during error logging
+            }
+        }
+
+        /// <summary>
+        /// Custom TextWriter that writes directly to the standard output/error streams
+        /// using native system calls. This is required because the standard Console
+        /// output redirection might not work as expected in the embedded environment
+        /// where the .NET runtime is hosted by SQL Server.
+        /// </summary>
+        private class InteropTextWriter : System.IO.TextWriter
```

Comment:
`InteropTextWriter` mixes platform detection and writing logic. 
Consider splitting into two platform-specific implementations for clarity.